test/test_torch.py

```diff
@@ -14166,6 +14166,24 @@ def test_cross_device_binary_ops(self, devices):
                 with self.assertRaisesRegex(RuntimeError, "Expected all tensors.+"):
                     op(cpu_tensor, a)
 
+    # This test ensures that a scalar Tensor can be safely used
+    # in a binary operation in conjuction with a Tensor on all
+    # available CUDA devices
+    @deviceCountAtLeast(2)
+    @onlyCUDA
+    def test_binary_op_scalar_device_unspecified(self, devices):
+        scalar_val = torch.tensor(1.)
+        for default_device in devices:
+            with torch.cuda.set_device(default_device):
```

Comment:
Does set_device work? I thought it is `torch.cuda.device(default_device)`.

Reply:
Yeah, it works, although the docs say "Usage of this function is discouraged in favor of `device`." Not sure why that is, but I'll change it.

Reply:
Oh you're right, `cuda.set_device` doesn't create a context manager.

Reply:
I'll wait for the tests, and should be good to merge.